app/controllers/provider_interface/notifications_controller.rb

```diff
@@ -1,11 +1,15 @@
 module ProviderInterface
   class NotificationsController < ProviderInterfaceController
     def update
-      current_provider_user.update!(
-        send_notifications: notification_params[:send_notifications],
-      )
-
       if FeatureFlag.active?(:configurable_provider_notifications)
+        notification_preferences = current_provider_user.notification_preferences
+        notification_preferences.assign_attributes(notification_preferences_params)
+
+        notification_preferences.save! if notification_preferences.valid?
```

Comment:
if this is invalid we won't save, but will display the success message. So we can either assume everything is valid and `save!` which will end up throwing an exception and indicating there is an issue, or handle the invalid model by re-rendering the update page and allowing for the errors to be rendered.